scripts/src/flair.js

```diff
@@ -0,0 +1,121 @@
+/*
+  Copyright 2019 OmiseGO Pte Ltd
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+  http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+const { get, find, orderBy, difference } = require('lodash')
+const fetch = require('node-fetch');
+const util = require('./util.js')
+
+module.exports = {
+  flairGetter: function ({ curr, burnAddr}, txs, purchaseVerifier) {
+    return function (flairName, flairText, price) {
+      const res = txs.reduce(function(owners, tx) {
+        const isValid = purchaseVerifier(tx, price, curr, burnAddr, flairName)
+        if(isValid) {
+          owners.push(util.getOwner(tx, curr))
+          return owners
+        }
+        return owners
+      }, [])
+      return { flairName: flairName, flairText: flairText, addresses: new Set(res) }
+    }
+  },
+
+  parseThreadJSON: function (json) {
+    const rawComments = get(json, '[1].data.children', []);
+
+    let userAddressMap = [];
+
+    // sanitize and filter comments for addresses
+    for (const comment of rawComments) {
+      const rawText = comment.data.body;
+      const words = rawText.trim().split(' ');
```

Comment:
added a fix here in the extension for not parsing the newlines so you just have to add
```
const words = rawText.trim().replace(/\n/g, ' ').split(' ');
```